airbyte-ci/connectors/pipelines/pipelines/steps/simple_ci_step.py

```diff
@@ -0,0 +1,115 @@
+            internal_tools (List[MountPath], optional): internal tools to install. Defaults to [].
+            secrets (dict[str, dagger.Secret], optional): secrets to add to container. Defaults to {}.
+            env_variables (dict[str, str], optional): env variables to set in container. Defaults to {}.
+            working_directory (str, optional): working director to run the command in. Defaults to "/".
+            command (Optional[List[str]], optional): The default command to run. Defaults to None.
+        """
+        self.title = title
+        super().__init__(context)
+
+        self.paths_to_mount = paths_to_mount
+        self.working_directory = working_directory
+        self.internal_tools = internal_tools
+        self.secrets = secrets
+        self.env_variables = env_variables
+        self.command = command
+
+    def _mount_paths(self, container: dagger.Container) -> dagger.Container:
+        for path_to_mount in self.paths_to_mount:
+            path_string = str(path_to_mount)
+            destination_path = f"/{path_string}"
+            if path_to_mount.is_file:
+                file_to_load = self.context.get_repo_file(path_string)
+                container = container.with_mounted_file(destination_path, file_to_load)
+            else:
+                container = container.with_mounted_directory(destination_path, self.context.get_repo_dir(path_string))
+        return container
+
+    async def _install_internal_tools(self, container: dagger.Container) -> dagger.Container:
+        for internal_tool in self.internal_tools:
+            container = await with_installed_pipx_package(self.context, container, str(internal_tool))
+        return container
+
+    def _set_workdir(self, container: dagger.Container) -> dagger.Container:
+        return container.with_workdir(self.working_directory)
+
+    def _set_env_variables(self, container: dagger.Container) -> dagger.Container:
+        for key, value in self.env_variables.items():
+            container = container.with_env_variable(key, value)
+        return container
+
+    def _set_secrets(self, container: dagger.Container) -> dagger.Container:
+        for key, value in self.secrets.items():
+            container = container.with_secret_variable(key, value)
+        return container
+
+    async def init_container(self) -> dagger.Container:
+        # TODO (ben): Replace with python base container when available
+        container = with_python_base(self.context)
+
+        container = self._mount_paths(container)
+        container = self._set_env_variables(container)
+        container = self._set_secrets(container)
+        container = await self._install_internal_tools(container)
+        container = self._set_workdir(container)
+
+        return container
+
+    async def _run(self, command=None) -> StepResult:
+        command_to_run = command or self.command
```

Comment:
This means you can pass it explicitly or via context, right? Do we plan on later deprecating the latter?

Reply:
Hmm I think more accurately its that you can either set the command at build or run time.

i.e. I can create a step using this class that can either
1. Only ever run one specific command 
2. Will run whatever command you give it at run time.

Can be useful if I want to change a steps arguments based on the output of a previous step

Reply:
ooooh. that makes sense! thanks. maybe `input_command` or something to differentiate between `self.command` would help here